go-controller/pkg/libovsdbops/portbinding.go

```diff
@@ -0,0 +1,71 @@
+package libovsdbops
+
+import (
+	"fmt"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/sbdb"
+)
+
+// GetPortBinding looks up a port binding from the cache
+func GetPortBinding(sbClient libovsdbclient.Client, portBinding *sbdb.PortBinding) (*sbdb.PortBinding, error) {
+	found := []*sbdb.PortBinding{}
+	opModel := operationModel{
+		Model:          portBinding,
+		ModelPredicate: func(item *sbdb.PortBinding) bool { return item.LogicalPort == portBinding.LogicalPort },
+		ExistingResult: &found,
+		ErrNotFound:    true,
+		BulkOp:         false,
+	}
+
+	m := newModelClient(sbClient)
+	err := m.Lookup(opModel)
+	if err != nil {
+		return nil, err
+	}
+
+	return found[0], nil
+}
+
+func UpdatePortBindingSetChassis(sbClient libovsdbclient.Client, portBinding *sbdb.PortBinding, chassis *sbdb.Chassis) error {
```

Comment:
ack this works for me and ti would be the best if this is done by OVN. In respect of time I'd like to see a:
TODO (numans): Move this logic to OVN and come back to remove this in OVNK :) note written here so that we follow this up.